src/vmm_mad/remotes/lib/lxd/client.rb

```diff
@@ -119,19 +119,24 @@ def socket(lxd_path)
     def get_response(request, data)
         request.body = JSON.dump(data) unless data.nil?
 
-        request.exec(@socket, '1.1', request.path)
-
         response = nil
 
-        loop do
-            response = Net::HTTPResponse.read_new(@socket)
-            break unless response.is_a?(Net::HTTPContinue)
-        end
+        5.times do
```

Comment:
* Use a class constant API_RETRY, then API_RETRY.times
* A break statement is missing to not perform uneeded retries
* If possible raise LXDError(`Failed to read response from LXD`)